src/webgpu/api/validation/capability_checks/limits/maxSampledTexturesPerShaderStage.spec.ts

```diff
@@ -83,18 +99,30 @@ g.test('createPipelineLayout,at_over')
     await t.testDeviceWithRequestedMaximumLimits(
       limitTest,
       testValueName,
-      async ({ device, testValue, shouldError }) => {
-        const kNumGroups = 3;
+      async ({ device, testValue, shouldError, actualLimit }) => {
+        const maxBindingsPerBindGroup = Math.min(
+          t.device.limits.maxBindingsPerBindGroup,
+          actualLimit
+        );
+        const kNumGroups = Math.ceil(testValue / maxBindingsPerBindGroup);
+
+        // Not sure what to do in this case but best we get notified if it happens.
+        assert(kNumGroups < t.device.limits.maxBindGroups);
```

Comment:
`assert(kNumGroups <= t.device.limits.maxBindGroups);` ? (and below)

Reply:
done! thanks for catching